src/Microsoft.OpenApi/Reader/V3/OpenApiV3Deserializer.cs

```diff
@@ -193,7 +193,11 @@ private static (string, string) GetReferenceIdAndExternalResource(string pointer
             var refId = refSegments.Last();
             var isExternalResource = !refSegments.First().StartsWith("#", StringComparison.OrdinalIgnoreCase);
 
-            string externalResource = isExternalResource ? $"{refSegments.First()}/{refSegments[1].TrimEnd('#')}" : null;
+            string externalResource = null;
+            if (isExternalResource)
+            {
+                externalResource = pointer.Split('#').FirstOrDefault()?.TrimEnd('#');
```

Comment:
```suggestion
                externalResource = pointer.Split('#')[0].TrimEnd('#');
```

`string.Empty.Split('a')` returns `[string.Empty]`